File: Sorting Algo/QuickSort.cpp
#include<iostream>
using namespace std;
int partion(int arr[],int low,int high)
{
    int pivot = arr[low];
    int i=low,j=high;

    do
    {
        do { i++;}while(arr[i]<=pivot);
        do { j--;}while(arr[j]>pivot);
        if(i<j) swap(arr[i],arr[j]);
    }while(i<j);
    swap(arr[j],arr[low]);
    return j;
}
void QuickSort(int arr[],int low,int high)
{
    if(low<high)
    {
        int j = partion(arr,low,high);
        QuickSort(arr,low,j);
        QuickSort(arr,j+1,high);
    }
}
int main()
{
    int arr[] = {7,9,6,1,3,5,10};
    int len = sizeof(arr)/sizeof(arr[0]);
    QuickSort(arr,0,len);
    for(int i=0;i<len;i++) cout<<arr[i]<<" ";
    return 0;
}

File: Sorting Algo/QuickSort.py
import random

# Function to generate random unique integers
def randomIn(n):
    arr = []
    i = 0
    while i < n:
        x = random.randint(1, 10)
        if x not in arr:
            arr.append(x)
            i += 1
    return arr

def partition(arr,low,high):
    pivot = arr[low]
    i = low
    j = high
    while i<j:
        i+=1
        while i<high and arr[i]<=pivot:
            i+=1
        j-=1
        while arr[j]>pivot:
            j-=1

        if(i<j):
            arr[i],arr[j] = arr[j],arr[i]
            
    arr[low],arr[j] = arr[j],arr[low]
    return j
    
def quicksort(arr,low,high):
    if low <high:
        pi = partition(arr,low,high)
        quicksort(arr,low,pi)
        quicksort(arr,pi+1,high)

# Main program
n = 5
arr = randomIn(n)
print("Unsorted array:", arr)

quicksort(arr, 0, n)
print("Sorted array:", arr)
